### PR TITLE
Added c5g7 benchmark problem for MG mode

### PR DESCRIPTION
**Pull request by nelsonag**
Forget the previous PR; I had some odd commits from 2013 apparently that made no sense.

### CODE REVIEW

**Comment by nelsonag** [comment]
Quick note: this isn't 100% ready for merge (and it makes no sense to do so anyways without #494 of openmc being merged first), I am also adding the c5g7 extended 3d benchmark and I want to cleanup the xml input files a bit.  I put this PR in a little early so a multi-group model would be available for the reviewers of #494.

**Comment by nelsonag** [comment]
Now this is ready.

**Comment by samuelshaner** [comment]
What do you think about just having the Python input files instead of xml to minimize code reuse? I've created some OpenMC Python input files for C5G7 here:

https://github.com/samuelshaner/openmc/tree/mg/examples/python/c5g7

**Comment by nelsonag** [comment]
I see no problem with either way

**Comment by nelsonag** [comment]
@paulromano is there a maintainer-preference for input format (i.e., python vs xml)?

**Comment by paulromano** [comment]
Using the Python API would be my preference.

**Comment by nelsonag** [comment]
I'm getting there, albeit slowly.  Comparing my results with those from @samuelshaner first to see if there any differences of note.

**Comment by nelsonag** [comment]
Ok, this is ready to go.  I used the python api input format from @samuelshaner, which I hope he doesnt mind.  

Here is a summary of results:

Case            | Reference Sol'n                                                       | OpenMC Sol'n | Error [pcm]
---------------- | ----------------------------------------------------------------|------------------- |---------------
UO2 Pin        | 1.32519(11) (From MCNP, not in official standard) | 1.32555(5)     | 27
2D                 | 1.18655(8)                                                              | 1.18662(11)   | -6
3D                 | 1.18381(8)                                                              | 1.18368(6)     | 11
3D unrodded | 1.14308(6)                                                              | 1.14311(6)    | -3
3D rodded-A | 1.12806(6)                                                               | 1.12811(6)    | -4
3D rodded-B | 1.07777(6)                                                               | 1.07767(6)    | 9

The diffrences are on the same order of, or significantly less than, the spread in eigenvalues published with the benchmarks for other MC MG codes (i.e., see table 3 of the original C5G7 spec).

The UO2 Pin problem is not in the official standard, but I've seen it in a few papers as the 'first problem' done so thought it worthwhile.  The reference solution above is from MCNP (MG) done by the MPACT guys (this # was from their validation files).  I cant speak to how right it is or is not.

**Comment by samuelshaner** [comment]
@nelsonag - the results look great! Glad to see you used the python input files.

File: c5g7/openmc/lattices.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+lattices['MOX Rodded Assembly'].dimension = [17, 17]
+lattices['MOX Rodded Assembly'].lower_left = [-10.71, -10.71]
+lattices['MOX Rodded Assembly'].pitch = [1.26, 1.26]
+m = universes['MOX 4.3%']
+n = universes['MOX 7.0%']
+o = universes['MOX 8.7%']
+r = universes['Control Rod']
+f = universes['Fission Chamber']
+lattices['MOX Rodded Assembly'].universes = [[m, m, m, m, m, m, m, m, m, m, m, m, m, m, m, m, m],
+                                               [m, n, n, n, n, n, n, n, n, n, n, n, n, n, n, n, m],
```
Comment (samuelshaner) [inline c5g7/openmc/lattices.py]: Can you remove the extra two spaces before rows 2+ in this lattice? This is probably an issue that I had in my inputs.

**Comment by samuelshaner** [comment]
Just tested all the inputs and everything checks out. I think this is ready to merge once the spacings is updated in lattices.py

**Comment by nelsonag** [comment]
Great, thanks for the quick review. I will get to the spacing tonight.

**Comment by nelsonag** [comment]
Fixed,

**Comment by paulromano** [comment]
Thanks @nelsonag (and @samuelshaner for reviewing)!

**Pull request merged**